core/ecschema-metadata/src/UnitProvider/SchemaUnitProvider.ts

```diff
@@ -17,13 +17,22 @@ import { UnitConverter } from "../UnitConversion/UnitConverter";
  */
 export class SchemaUnitProvider implements UnitsProvider {
   private _unitConverter: UnitConverter;
+  private _context: SchemaContext;
+
   /**
    *
-   * @param _context SchemaContext that contains the Units from which querying takes place.
+   * @param contextOrLocater The SchemaContext that contains the Units schema or the ISchemaLocater used to locate the schema. If the later, a new
+   * SchemaContext will be created and the locater will be added to the context.
    * @param _unitExtraData Additional data like alternate display label not found in Units Schema to match with Units; Defaults to empty array.
    */
-  constructor(private readonly _context: SchemaContext, private _unitExtraData: UnitExtraData[] = []) {
-    this._unitConverter = new UnitConverter(_context);
+  constructor(contextOrLocater: SchemaContext|ISchemaLocater, private _unitExtraData: UnitExtraData[] = []){
```

Comment:
Any reason  for an or here since a SchemaContext is a ISchemaLocater?

Reply:
Yeah, I updated the docs.  I thinks its clear(?) now.  Let me know what you think.

Reply:
I think it's good